pandas/core/indexes/multi.py

```diff
@@ -1783,16 +1792,23 @@ def is_lexsorted(self):
     @cache_readonly
     def lexsort_depth(self):
         if self.sortorder is not None:
-            if self.sortorder == 0:
-                return self.nlevels
-            else:
-                return 0
+            return self.sortorder
+
+        return self._lexsort_depth()
 
+    def _lexsort_depth(self):
```

Comment:
right was wanting a 
```def _lexsort_depth(self) -> int:``` here

Reply:
Ok sorry i did'nt get the "hint".
Joke aside, is it related to the caching of the result that the function needs typing?